clang/lib/CIR/Dialect/Transforms/LoweringPrepare.cpp

```diff
@@ -1061,6 +1077,25 @@ void LoweringPreparePass::lowerIterEndOp(IterEndOp op) {
   op.erase();
 }
 
+void LoweringPreparePass::buildGlobalAnnotationValues() {
+  if (globalAnnotations.empty())
+    return;
+  std::vector<mlir::Attribute> annotationValueVec;
+
+  for (auto &annotEntry : globalAnnotations) {
+    auto annot = dyn_cast<mlir::cir::AnnotationAttr>(annotEntry.first);
```

Comment:
if you are using `dyn_cast` I'd expect some test or assert for the case it fails, if you are sure these are `AnnotationAttr` it should just be a `cast`.